mlir-tensorrt/compiler/include/mlir-tensorrt/Compiler/OptionsRegistry.h

```diff
@@ -71,14 +72,15 @@ optionsCreateFromArgs(const CompilerClient &client,
         llvm::iterator_range(args), err);
   }
 
-  // TODO: Figure out whether to add a method in the base class like
-  // "finalizeOptions" or a callback here, or something else if
-  // `inferDeviceOptionsFromHost` is unique to StableHLO.
-  //
-  // Populate device options from host information.
-  Status inferStatus = result->inferDeviceOptionsFromHost();
-  if (!inferStatus.isOk())
-    return inferStatus;
+  llvm::Error finalizeStatus = result->finalize();
+
+  std::optional<std::string> errMsg{};
+  llvm::handleAllErrors(
+      std::move(finalizeStatus),
+      [&errMsg](const llvm::StringError &err) { errMsg = err.getMessage(); });
+
+  if (errMsg)
+    return getInternalErrorStatus(errMsg->c_str());
```

Comment:
Prefer to add a bit more context. I think you can use printf-style formatting with this function:

```
return getInternalErrorStatus("failed to initialize options: %s", errMsg->c_str());
```

Reply:
Done